lib/robots/dor_repo/speech_to_text/stt_create.rb

```diff
@@ -11,11 +11,35 @@ def initialize
 
         # available from LyberCore::Robot: druid, bare_druid, workflow_service, object_client, cocina_object, logger
         def perform_work
-          # TODO: do the caption creation by calling out to Whisper
+          # start the caption creation by sending an SQS message to the caption creation service
+          send_sqs_message
 
           # Leave this step running until the Whisper monitoring job marks it as complete
           LyberCore::ReturnState.new(status: :noop, note: 'Initiated SpeechToText.')
         end
+
+        private
+
+        def aws_provider
+          @aws_provider ||= Dor::TextExtraction::AwsProvider.new(region: Settings.aws.region, access_key_id: Settings.aws.access_key_id, secret_access_key: Settings.aws.secret_access_key)
+        end
+
+        def send_sqs_message
+          # Define the queue URL and message body
+          queue_url = Settings.aws.sqs_queue_url
+          message_body = {
```

Comment:
The body of the message that speech-to-text is currently looking for is documented here:

https://github.com/sul-dlss/speech-to-text/blob/docker/README.md#the-job-file

Reply:
So is the idea that we need to send one message per file that we want transcribed?  if so, then we can iterate over the druid folder in S3, find all of the files, and send multiple messages

though then...how would we know when all of the files in a given druid are done?

Reply:
oh, hmm, that's a good point.  maybe that's an argument for having the speech-to-text service do a directory or a list of files, instead of one file at a time.

alternatively, i can vaguely imagine some ways that the watcher could figure out whether a particular success/failure message for one file, possibly among many for a druid version, is the last one being processed, allowing the step to get marked completed if there were no errors in the batch:
* we could use WFS vars, with e.g. a hash of files to process by the wf for the druid version (hash initialized in this wf step), the processing status updated for each file as it's processed (set upon SQS message receipt by the [watcher daemon we've been discussing today](https://github.com/sul-dlss/common-accessioning/issues/1358)).  and then each time an update is received, the final success/error state of the step could be set?
* we could give common-accessioning its own data store to track that info, but otherwise use similar logic.

i'd probably vote against adding a database to common-accessioning, with no preference between the other two options (and interested in other better options if anyone can think of any).

adding extra load/responsibility to WFS seems not great, but this also seems along the lines of what WFS variables are meant for.  and it doesn't seem like it'd be adding that much load, if we only have a small handful of worker containers processing STT jobs.

keeping the speech-to-text worker container simple also seems desirable, but processing a list of files instead of one file isn't a terrible extra complication (sorry for the back and forth if we do go that route @edsu, i feel like we talked you into one file at a time when we all initially met to look at your work on https://github.com/sul-dlss/speech-to-text/pull/9).

Reply:
given the bare-bones nature of both our argo GUI and workflow-server-rails/lyber-core CLI tools for interacting with robots and workflow state, i'd guess that we shouldn't try to make individual media file extraction failures retriable, and that we should just let the whole batch for the druid version fail if one file fails, then fix the issue and retry the step if need be.  if we run into a bunch of obviously automatically retriable errors, maybe we can revisit and add retry handling for those things where ever is most appropriate?

Reply:
another thought, which could be an optimization: upon successful text generation for a given media file, the watcher process downloads the output for that media file.  then the speech-to-text job queuing code in this class could look to see if there is any output already, and only queue jobs for media files requiring STT that don't already have output.  that saves needless re-running and and thus GPU waste, without having to think about trickier retry capabilities.

Reply:
we'll might need to update https://github.com/sul-dlss/common-accessioning/issues/1358 with whatever comes out of this thread.

Reply:
oh, heh, i may have written all that for nothing: looking back again at the README in https://github.com/sul-dlss/speech-to-text/pull/9/, and at the `download_media` code in that PR for the first time (sort of reviewing these two in parallel, and occasionally flipping over to #1350)... it seems like it already handles a list with multiple files, since the `media` field in the job structure is an array.

i guess the "download all the successful extractions in a partially failed batch" even if it the whole batch doesn't succeed is still a useful thought, and i might've gotten there faster for all the likely useless writing above that 😆

Reply:
also, `SecureRandom.uuid` for the job ID?

Reply:
Thanks for summarizing.